<!-- Usage: roslaunch apc grasp_annotator.launch object_name:=expo_dry_erase_board_eraser keep_old_grasps_bool:=true -->
<launch>
    <!-- Arguments initialization -->
    <arg name="object_marker" value="object_marker" />
    <arg name="gripper_marker" value="gripper_marker" />
    <arg name="object_moves_topic" value="object_moves" />
    <arg name="gripper_moves_topic" value="gripper_moves" />
    <arg name="gripper_width_topic" value="gripper_width" />
    <arg name="update_rate" value="50" />

    <!-- Handling for object_name argument -->
    <arg name="object_name" />

    <!-- Handling for keep_old_grasps argument -->
    <arg name="keep_old_grasps_bool" default="false" />
    <arg name="keep_old_grasps" value="" unless="$(arg keep_old_grasps_bool)"/>
    <arg name="keep_old_grasps" value="-k" if="$(arg keep_old_grasps_bool)" />

    <!-- APC nodes in apc namespace -->
    <group ns="apc">
        <!-- Node for object marker -->
        <node name="$(arg object_marker)"
              pkg="apc"
              type="rviz_marker_publisher.py"
              args="-n object_marker -t 1 -d -u 50 -c $(arg object_moves_topic) -m data/meshes/$(arg object_name).stl" />

        <!-- Node for gripper marker -->
        <node name="$(arg gripper_marker)"
              pkg="apc"
              type="rviz_marker_publisher.py"
              args="-n gripper_marker -t 2 -d -u 50 -c $(arg gripper_moves_topic) -g $(arg gripper_width_topic)" />

        <!-- Node for grasp annotator -->
        <node name="grasp_annotator"
              pkg="apc"
              type="rviz_grasp_annotator.py"
              args="--object-name $(arg object_name) --object-marker $(arg object_marker) --gripper-marker $(arg gripper_marker) --object-moves-topic $(arg object_moves_topic) --gripper-moves-topic $(arg gripper_moves_topic) --gripper-width-topic $(arg gripper_width_topic) --update-rate $(arg update_rate) $(arg keep_old_grasps)" />
    </group>
</launch>
